docs/src/aleph_bft_api.md

```diff
@@ -2,19 +2,28 @@
 
 ### 3.1 Required Trait Implementations.
 
-#### 3.1.1 DataIO.
+#### 3.1.1 DataProvider & FinalizationHandler.
 
-The DataIO trait is an abstraction for a component that provides data items and allows to input ordered data items. `DataIO` is parametrized with a `Data` generic type representing the type of items we would like to order. Below we give examples of what these might be.
+The DataProvider trait is an abstraction for a component that provides data items. `DataProvider` is parametrized with a `Data` generic type representing the type of items we would like to order. Below we give examples of what these might be.
 
 ```rust
-pub trait DataIO<Data> {
-    type Error: Debug;
+pub trait DataProvider<Data> {
     fn get_data(&self) -> Data;
```

Comment:
Similarly to previous comment: consider changing to `&mut self` so at least tests and examples doesn't need to use `Mutex/Cell` or similar.

Reply:
added